src/main/java/com/nha/abdm/wrapper/hip/hrp/database/mongo/services/RequestLogService.java

```diff
@@ -356,28 +355,32 @@ public void persistConsentInitRequest(
     mongoTemplate.save(requestLog);
   }
 
-  public void updateConsentResponse(String requestId, String consentRequestId) {
-    Query query = new Query(Criteria.where(FieldIdentifiers.GATEWAY_REQUEST_ID).is(requestId));
-    Map<String, Object> map = new HashMap<>();
-    map.put(FieldIdentifiers.CONSENT_REQUEST_ID, consentRequestId);
-    Update update = new Update();
-    update.set(FieldIdentifiers.RESPONSE_DETAILS, map);
-    update.set(FieldIdentifiers.STATUS, RequestStatus.CONSENT_ON_INIT_RESPONSE_RECEIVED);
-    mongoTemplate.updateFirst(query, update, RequestLog.class);
-  }
+  //  public void updateConsentResponse(String requestId, String consentRequestId) {
+  //    Query query = new Query(Criteria.where(FieldIdentifiers.GATEWAY_REQUEST_ID).is(requestId));
+  //    Map<String, Object> map = new HashMap<>();
+  //    map.put(FieldIdentifiers.CONSENT_REQUEST_ID, consentRequestId);
+  //    Update update = new Update();
+  //    update.set(FieldIdentifiers.RESPONSE_DETAILS, map);
+  //    update.set(FieldIdentifiers.STATUS, RequestStatus.CONSENT_ON_INIT_RESPONSE_RECEIVED);
+  //    mongoTemplate.updateFirst(query, update, RequestLog.class);
+  //  }
 
-  public void updateConsentResponse(String requestId, ConsentStatus consentDetails)
+  public <T> void updateConsentResponse(
```

Comment:
Good to have: Is it a good idea to have some indicator in the `RequestLog` table saying what is the request about? like `CONSENT_CREATION`, `CARE_CONTEXT_LINKING`, `DATA_TRANSFER_FETCH` etc?

Reply:
we can have that to enable hinting or if there is a concrete requirement like what we will do with this hint or metadata?

Reply:
Have created an issue to simplify this class: https://github.com/NHA-ABDM/ABDM-wrapper/issues/76